src/handles/tpm.rs

```diff
@@ -0,0 +1,181 @@
+                tss_tpm_handle,
+            )?)),
+            TPM2_HT_TRANSIENT => Ok(TpmHandle::Transient(TransientTpmHandle::new(
+                tss_tpm_handle,
+            )?)),
+            TPM2_HT_PERSISTENT => Ok(TpmHandle::Persistent(PersistentTpmHandle::new(
+                tss_tpm_handle,
+            )?)),
+            TPM2_HT_AC => Ok(TpmHandle::Ac(AcTpmHandle::new(tss_tpm_handle)?)),
+            _ => {
+                error!("Error: Invalid TPM handle type {}", most_significant_byte);
+                Err(Error::local_error(WrapperErrorKind::InvalidParam))
+            }
+        }
+    }
+}
+
+/// Macro for creating the specific TPM handle types
+macro_rules! create_tpm_handle_type {
+    ($handle_type_name:ident, $tpm_handle_kind:path, $tpm_handle_type_id:tt) => {
+        #[derive(Debug, Copy, Clone, Eq, PartialEq)]
+        pub struct $handle_type_name {
+            value: u32,
+        }
+
+        impl $handle_type_name {
+            pub fn new(value: u32) -> Result<$handle_type_name> {
+                if value.to_be_bytes()[0] != $tpm_handle_type_id {
+                    error!(
+                        "Errro: TPM Handle ID of the input value did not match the {} (!={})",
+                        stringify!($handle_type_name),
+                        $tpm_handle_type_id
+                    );
+                    return Err(Error::local_error(WrapperErrorKind::InvalidParam));
+                }
+                Ok($handle_type_name { value })
+            }
+        }
+
+        impl From<$handle_type_name> for TpmHandle {
+            fn from(specific_tpm_handle: $handle_type_name) -> TpmHandle {
+                $tpm_handle_kind(specific_tpm_handle)
+            }
+        }
+
+        impl TryFrom<TpmHandle> for $handle_type_name {
+            type Error = Error;
+            fn try_from(general_tpm_handle: TpmHandle) -> Result<$handle_type_name> {
+                match general_tpm_handle {
+                    $tpm_handle_kind(val) => Ok(val),
+                    _ => {
+                        error!(
+                            "Error: Incorrect tpm handle kind(=={}) when converting to {}",
+                            stringify!($tpm_handle_kind),
+                            stringify!($handle_type_name)
+                        );
+                        return Err(Error::local_error(WrapperErrorKind::InvalidParam));
+                    }
+                }
+            }
+        }
+
+        impl From<$handle_type_name> for TPM2_HANDLE {
+            fn from(specific_tpm_handle: $handle_type_name) -> TPM2_HANDLE {
+                specific_tpm_handle.value
+            }
+        }
+
+        impl TryFrom<TPM2_HANDLE> for $handle_type_name {
+            type Error = Error;
+            fn try_from(tss_tpm_handle: TPM2_HANDLE) -> Result<$handle_type_name> {
+                $handle_type_name::new(tss_tpm_handle)
+            }
+        }
+    };
+}
```

Comment:
Nice, this is exactly what I've just done for Buffer as well, shall push a PR today